internal/fs/fs.go

```diff
@@ -976,6 +976,13 @@ func (fs *fileSystem) lookUpOrCreateChildDirInode(
 func (fs *fileSystem) syncFile(
 	ctx context.Context,
 	f *inode.FileInode) (err error) {
```

Comment:
The call shouldn't reach here if the inode is not valid. In what scenario does it reach here?

Reply:
file.close() on local file makes a FlushFile call.

Reply:
1. unlinking a synced file
  The calls made to gcsfuse are
    ```
    Unlink()
    SetInodeAttributes()
    LookUpInode()
    ForgetInode()
    ```

2. process 1: hold a file handle f.
 process 2: unlink the file from another process
 process 1: f.close()
    The calls made to gcsfuse are
    ```
    Unlink()
    SetInodeAttributes()
    LookUpInode()
    FlushFile()
    ReleaseFileHandle()
    ForgetInode()
    ```

2. process 1: hold a file handle f.
 process 2: unlink the file from another process
 process 1: f.write("content")
 process 1: f.close()
    The calls made to gcsfuse are
    ```
    Unlink()
    SetInodeAttributes()
    LookUpInode()
    WriteFile() -> returns Error: "no such file or directory"
    SetInodeAttributes()
    FlushFile()
    ReleaseFileHandle()
    ForgetInode()
    ```
Please also take a look at the updated PR description as well.

Reply:
Thanks for the details Ashmeen. Like i mentioned in earlier writeFlow will not always throw an error.